msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerAuthServiceStrategy.java

```diff
@@ -87,161 +93,142 @@ Intent getBrokerAuthorizationIntent(@NonNull final AcquireTokenOperationParamete
         return interactiveRequestIntent;
     }
 
-    private Intent getBrokerAuthorizationIntentFromAuthService(@NonNull final AcquireTokenOperationParameters parameters)
-            throws ClientException {
-        final String methodName = ":getBrokerAuthorizationIntentFromAuthService";
-        Telemetry.emit(
-                new BrokerStartEvent()
-                        .putAction(methodName)
-                        .putStrategy(TelemetryEventStrings.Value.BOUND_SERVICE)
-        );
-
-        IMicrosoftAuthService service;
-        Intent resultIntent;
-
-        final MicrosoftAuthClient client = new MicrosoftAuthClient(parameters.getAppContext());
-        final MicrosoftAuthServiceFuture authServiceFuture = client.connect();
+    /**
+     * A broker task to be performed. Use in conjunction with performBrokerTask()
+     */
+    public interface AuthServiceOperation<T> {
+
+        /**
+         * Performs a task in this function with the given IMicrosoftAuthService.
```

Comment:
nit: Changed `function` to `method`

also might be good to link to the `IMicrosoftAuthService` class in the doc